src/OpenSpaceToolkit/Astrodynamics/Trajectory/Segment.cpp

```diff
@@ -584,38 +710,233 @@ const NumericalSolver& Segment::accessNumericalSolver() const
+        std::make_shared<HeterogeneousGuidanceLaw>();
+    const auto acceptManeuver = [&](const Segment::Solution& maneuverSolution, const Interval& maneuverInterval) -> void
+    {
+        BOOST_LOG_TRIVIAL(debug) << "Maneuver accepted: " << maneuverInterval.toString() << std::endl;
+
+        segmentStates.add(Array<State>(maneuverSolution.states.begin() + 1, maneuverSolution.states.end()));
+
+        segmentConditionIsSatisfied = maneuverSolution.conditionIsSatisfied;
+
+        previousManeuverInterval = maneuverInterval;
+
+        segmentHeterogenousGuidanceLaw->addGuidanceLaw(
+            maneuverSolution.getThrusterDynamics()->getGuidanceLaw(), maneuverInterval
+        );
+    };
+
+    // Helper lambda to handle maximum duration violation strategies, returning true if the segment condition is
+    // satisfied. It also updates the segment states and the guidance law if the maneuver is accepted.
+    const auto handleMaximumDurationViolation = [&](const Interval& candidateManeuverInterval) -> bool
+    {
+        switch (maneuverConstraints_.maximumDurationStrategy)
+        {
+            case MaximumManeuverDurationViolationStrategy::Fail:
+            {
+                throw ostk::core::error::RuntimeError(
+                    "Maneuver duration exceeds maximum maneuver duration constraint, change the maximum maneuver "
+                    "duration strategy to prevent the Sequence from failing."
+                );
+            }
+
+            case MaximumManeuverDurationViolationStrategy::Skip:
+            {
+                return solveAndAddCoastSegment(candidateManeuverInterval.getEnd());
+            }
+
+            case MaximumManeuverDurationViolationStrategy::LeadingSlice:
+            {
+                const Interval validManeuverInterval = Interval::Closed(
+                    candidateManeuverInterval.getStart(),
+                    candidateManeuverInterval.getStart() + maneuverConstraints_.maximumDuration
+                );
+
+                const Shared<Thruster> slicedThruster =
+                    buildThrusterDynamicsWithManeuverIntervals(validManeuverInterval);
+                const auto [maneuverSolution, maneuvers] = solveManeuver(slicedThruster);
+
+                acceptManeuver(maneuverSolution, validManeuverInterval);
+
+                return maneuverSolution.conditionIsSatisfied;
+            }
+
+            case MaximumManeuverDurationViolationStrategy::TrailingSlice:
+            {
+                const Interval validManeuverInterval = Interval::Closed(
+                    candidateManeuverInterval.getEnd() - maneuverConstraints_.maximumDuration,
+                    candidateManeuverInterval.getEnd()
+                );
+                const Shared<Thruster> slicedThruster =
+                    buildThrusterDynamicsWithManeuverIntervals(validManeuverInterval);
+                const auto [maneuverSolution, maneuvers] = solveManeuver(slicedThruster);
```

Comment:
Since these are not used, is there a way to just not set them in cpp like we do in python?
```suggestion
                const auto [maneuverSolution, ~] = solveManeuver(slicedThruster);
```